js/src/redux/providers/signerMiddleware.js

```diff
@@ -49,23 +51,56 @@ export default class SignerMiddleware {
   }
 
   onConfirmStart = (store, action) => {
-    const { id, password } = action.payload;
-
-    this._api.signer
-      .confirmRequest(id, {}, password)
-      .then((txHash) => {
-        console.log('confirmRequest', id, txHash);
-        if (!txHash) {
-          store.dispatch(actions.errorConfirmRequest({ id, err: 'Unable to confirm.' }));
-          return;
-        }
+    const { id, password, wallet, payload } = action.payload;
+
+    const handlePromise = promise => {
+      promise
+        .then((txHash) => {
+          console.log('confirmRequest', id, txHash);
+          if (!txHash) {
+            store.dispatch(actions.errorConfirmRequest({ id, err: 'Unable to confirm.' }));
+            return;
+          }
+
+          store.dispatch(actions.successConfirmRequest({ id, txHash }));
+        })
+        .catch((error) => {
+          console.error('confirmRequest', id, error);
+          store.dispatch(actions.errorConfirmRequest({ id, err: error.message }));
+        });
+    };
 
-        store.dispatch(actions.successConfirmRequest({ id, txHash }));
-      })
-      .catch((error) => {
-        console.error('confirmRequest', id, error);
-        store.dispatch(actions.errorConfirmRequest({ id, err: error.message }));
-      });
+    // Sign request in-browser
+    if (wallet && payload.transaction) {
+      const { transaction } = payload;
+      this._api.parity
+        .nextNonce(transaction.from)
```

Comment:
Is `Promise.resolve(...).then` executed in next even loop tick? The problem is that wallet derivation takes significant amount of time and UI is not refreshed before this happens. So it needs to be wrapped with `setTimeout(..., 0)`. Will try that and see if it works.